internal/control_test_engine/gnb/ngap/handler.go

```diff
@@ -820,33 +840,39 @@ func HandlerAmfStatusIndication(amf *context.GNBAmf, gnb *context.GNBContext, me
 				unavailableSetId := unavailableGuamiItem.GUAMI.AMFSetID.Value
 				unavailablePointer := unavailableGuamiItem.GUAMI.AMFPointer.Value
 
-				for i, amf := range context.GNBAmfList {
-					for j := 0; j < amf.GetLenPlmns(); j++ {
-						amfSupportMcc, amfSupportMnc := amf.GetPlmnSupport(j)
-						amfSupportRegionId := amf.GetRegionId()
-						amfSupportSetId := amf.GetSetId()
-						amfSupportPointer := amf.GetPointer()
-						log.Debugf("amf Support: %s %s %s %s %s\n", amfSupportMcc, amfSupportMnc,
-							ngapConvert.BitStringToHex(&amfSupportRegionId),
-							ngapConvert.BitStringToHex(&amfSupportSetId),
-							ngapConvert.BitStringToHex(&amfSupportPointer))
+				amfPool := gnb.GetAmfPool()
+				amfPool.Range(func(k, v any) bool {
+					oldAmf, ok := v.(*context.GNBAmf)
+					if !ok {
+						return true
+					}
+					for j := 0; j < oldAmf.GetLenPlmns(); j++ {
+						oldAmfSupportMcc, oldAmfSupportMnc := oldAmf.GetPlmnSupport(j)
+						oldAmfSupportRegionId := oldAmf.GetRegionId()
+						oldAmfSupportSetId := oldAmf.GetSetId()
+						oldAmfSupportPointer := oldAmf.GetPointer()
+						log.Debugf("amf Support: %s %s %s %s %s\n", oldAmfSupportMcc, oldAmfSupportMnc,
+							ngapConvert.BitStringToHex(&oldAmfSupportRegionId),
+							ngapConvert.BitStringToHex(&oldAmfSupportSetId),
+							ngapConvert.BitStringToHex(&oldAmfSupportPointer))
 						log.Debugf("unavailable: %s %s %s %s %s\n", unavailableMcc, unavailableMnc,
 							ngapConvert.BitStringToHex(&unavailableRegionId),
 							ngapConvert.BitStringToHex(&unavailableSetId),
 							ngapConvert.BitStringToHex(&unavailablePointer))
 
-						if amfSupportMcc == unavailableMcc && amfSupportMnc == unavailableMnc &&
-							compareAperBitString(&amfSupportRegionId, &unavailableRegionId) &&
-							compareAperBitString(&amfSupportSetId, &unavailableSetId) &&
-							compareAperBitString(&amfSupportPointer, &unavailablePointer) {
-							log.Info("[GNB][AMF] Remove AMF:", amf.GetAmfName())
-							tnla := amf.GetTNLA()
+						if oldAmfSupportMcc == unavailableMcc && oldAmfSupportMnc == unavailableMnc &&
+							compareAperBitString(&oldAmfSupportRegionId, &unavailableRegionId) &&
```

Comment:
Would Reflect.DeepEqual work as well instead of creating a new comparison function?

Reply:
Yes, it works well. I will update to `reflect.DeepEqual()`